pkg/services/publicdashboards/service/service.go

```diff
@@ -428,9 +428,21 @@ func GenerateAccessToken() (string, error) {
 }
 
 func (pd *PublicDashboardServiceImpl) newCreatePublicDashboard(ctx context.Context, dto *SavePublicDashboardDTO) (*PublicDashboard, error) {
-	uid, err := pd.NewPublicDashboardUid(ctx)
-	if err != nil {
-		return nil, err
+	var uid string
+	var err error
+	//Check if uid already exists, if none then auto generate
+	existingPubdash, _ := pd.store.Find(ctx, dto.PublicDashboard.Uid)
+	if existingPubdash != nil {
+		return nil, ErrPublicDashboardUidExists.Errorf("Create: public dashboard uid %s already exists", dto.Uid)
```

Comment:
```suggestion
		return nil, ErrPublicDashboardUidExists.Errorf("Create: public dashboard uid %s already exists", dto.PublicDashboard.Uid)
```
`dto.Uid` doesn't seem to be set. And the `store.Find` is made based on `dto.PublicDashboard.Uid`. I think we should use `dto.PublicDashboard.Uid`

Reply:
ah yes, that was a mistake on my end, you are correct